obspy/io/csv/core.py

```diff
@@ -0,0 +1,484 @@
+}
+PFIELDS = {
+    'basic': '{seedid} {phase} {time:.5f} {weight:.3f}'
+}
+CSZ_COMMENT = f'CSZ format v{__version__} obspy_no_uncompress'.encode('utf-8')
+# for load_csv
+DTYPE = {
+    'time': 'datetime64[ms]',
+    'lat': float,
+    'lon': float,
+    'dep': float,
+    'mag': float,
+    'magtype': 'U10',
+    'id': 'U50'
+}
+
+# catalog and contribid are not used
+EVENTTXT_NAMES = (  # for reading
+    'id time lat lon dep author catalog contrib contribid '
+    'magtype mag magauthor region')
+EVENTTXT_HEADER = (  # for writing
+    '#EventID | Time | Latitude | Longitude | Depth/km | Author | Catalog | '
+    'Contributor | ContributorID | MagType | Magnitude | MagAuthor | '
+    'EventLocationName')
+
+
+def _is_csv(fname, **kwargs):
+    try:
+        return _read_csv(fname, format_check=True)
+    except Exception:
+        return False
+
+
+def _is_eventtxt(fname, **kwargs):
+    try:
+        return _read_eventtxt(fname, format_check=True)
+    except Exception:
+        return False
+
+
+def _is_csz(fname, **kwargs):
+    try:
+        assert zipfile.is_zipfile(fname)
+        with zipfile.ZipFile(fname) as zipf:
+            assert (zipf.comment.startswith(b'CSZ') and
+                    zipf.comment.endswith(b'obspy_no_uncompress'))
+        return True
+    except Exception:
+        return False
+
+
+def _evid(event):
+    return str(event.resource_id).split('/')[-1]
+
+
+def _origin(event):
+    return event.preferred_origin() or event.origins[0]
+
+
+@contextmanager
+def _open(filein, *args, **kwargs):
+    """Accept both files or file names"""
+    if isinstance(filein, str):  # filename
+        with open(filein, *args, **kwargs) as f:
+            yield f
+    else:  # file-like object
+        yield filein
+
+
+def _names_sequence(names):
+    if isinstance(names, dict):
+        names = [names.get(i, '_') for i in range(max(names.keys())+1)]
+    elif ' ' in names:
+        names = names.split()
+    return names
+
+
+def _string(row, key):
+    if key in row and row[key].strip() != '':
+        return row[key].strip()
+
+
+def _read_csv(fname, skipheader=0, default=None, names=None,
+              check_compression=None, format_check=False,
+              **kwargs):
+    """
+    Read a CSV file and return ObsPy Catalog
+
+    :param skipheader: skip first rows of file
+    :param depth_in_kim: depth is given in kilometer (default: True) or
+        meter (False)
+    :param default: dictionary with default values, at the moment only
+         magtype is supported,
+         i.e. to set magtypes use `default={'magtype': 'Ml'}`
+    :param names: determined automatically from header line of file,
+        otherwise can be specified as string, sequence or dict
+    :param check_compression: not used by read_csv
+    :param format_check: Read only the first event
+    :param **kargs: all other kwargs are passed to csv.DictReader,
+        important additional arguments are fieldnames, dialect, delimiter, etc
```

Comment:
Check these params, they don't match up with the function signature (no fname, in params, no depth_in_km in signature)